lib/classes/TradeOffer.js

```diff
@@ -263,7 +262,7 @@ function addItem(details, offer, list) {
 		assetid: (details.assetid || details.id).toString(), // always needs to be a string
 		appid: parseInt(details.appid, 10), // always needs to be an int
 		contextid: details.contextid.toString(), // always needs to be a string
-		amount: parseInt(details.amount || 1, 10), // always needs to be an int
+		amount: parseInt(details.amount || 1, 10) // always needs to be an int
```

Comment:
Overcomplicating things imo. `parseInt` is fine. 
The `source` you linked shows both `parseInt(num, 10)` and `num << 0` as the same, or even favors `parseInt` for me.